test/e2e/play_kube_test.go

```diff
@@ -2498,6 +2567,83 @@ var _ = Describe("Podman kube play", func() {
 		Expect(inspect.OutputToString()).To(ContainSubstring("running"))
 	})
 
+	// If always restart init container didn't define define startup probe, it is started immediately and the Pod is starting regular container
+	It("test with always restart init container without startup probe started immediately", func() {
+		pod := getPod(withPodInitCtr(getCtr(withImage(CITEST_IMAGE), withCmd([]string{"top"}), withRestartableInitCtr(), withName("sidecar-container"))), withCtr(getCtr(withImage(CITEST_IMAGE), withCmd([]string{"top"}))))
+		err := generateKubeYaml("pod", pod, kubeYaml)
+		Expect(err).ToNot(HaveOccurred())
+
+		kube := podmanTest.Podman([]string{"kube", "play", kubeYaml})
+		kube.WaitWithDefaultTimeout()
+		Expect(kube).Should(ExitCleanly())
+
+		// Expect the number of containers created to be 3, infra, restartable init container and regular container
+		numOfCtrs := podmanTest.NumberOfContainers()
+		Expect(numOfCtrs).To(Equal(3))
+
+		// Init container should keep running
```

Comment:
Maybe you should first test that the main container is running and only then check the init container. This will show that the main was started without the init finishing.

Reply:
Do you mean changing the checking order, i.e., checking main container first and then init container?

I updated the order and force-pushed again. Please help check

Reply:
Yes, that's what I meant. Sorry if I wasn't clear enough